components/TransactionForm.js

```diff
@@ -0,0 +1,346 @@
+
+  &:hover {
+    background-color: var(--light-bg-color);
+  }
+
+  &:focus {
+    border-color: var(--accent-color);
+    outline: none;
+    box-shadow: 0 0 4px rgba(70, 134, 205, 0.8);
+  }
+`;
+
+const StyledCategoryLabel = styled.label`
+  grid-area: categoryLabel;
+`;
+
+const StyledCategorySelect = styled.select`
+  grid-area: categoryInput;
+  padding: 4px 12px;
+  border: 1px solid var(--dark-grey-color);
+  border-radius: 24px;
+  background-color: white;
+  color: var(--text-color-dark);
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
+  transition: all 0.2s ease-in-out;
+
+  &:hover {
+    background-color: var(--light-bg-color);
+  }
+
+  &:focus {
+    border-color: var(--accent-color);
+    outline: none;
+    box-shadow: 0 0 4px rgba(70, 134, 205, 0.8);
+  }
+`;
+
+const StyledTypeLabel = styled.label`
+  grid-area: typeLabel;
+`;
+
+const StyledRadioInput = styled.input`
+  grid-area: typeInput;
+`;
+
+const StyledRadioLabel = styled.label`
+  grid-area: radioLabel;
+`;
+
+const ErrorMessage = styled.p`
+  grid-area: typeErrorMessage;
+  color: red;
+  font-size: 10px;
+  margin-bottom: 4px;
+`;
+
+const StyledDateLabel = styled.label`
+  grid-area: dateLabel;
+`;
+
+const StyledDateInput = styled.input`
+  grid-area: dateInput;
+  padding: 4px 12px;
+  border: 1px solid var(--dark-grey-color);
+  border-radius: 24px;
+  background-color: white;
+  color: var(--text-color-dark);
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
+  transition: all 0.2s ease-in-out;
+
+  &:hover {
+    background-color: var(--light-bg-color);
+  }
+
+  &:focus {
+    border-color: var(--accent-color);
+    outline: none;
+    box-shadow: 0 0 4px rgba(70, 134, 205, 0.8);
+  }
+`;
+
+const StyledLegend = styled.legend`
+  padding: 6px;
+  text-align: center;
+  font-weight: 500;
+`;
+
+const StyledToggleButton = styled.div`
+  display: flex;
+  justify-content: space-around;
+  align-items: center;
+  font-size: 16x;
+  height: 24px;
+  background-color: var(--light-bg-color);
+  border: 1px solid var(--dark-grey-color);
+  border-radius: 30px;
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
+
+  input {
+    display: none;
+  }
+
+  label {
+    cursor: pointer;
+    border-radius: 30px;
+    width: 120px;
+    text-align: center;
+    color: var(--text-color-dark);
+  }
+
+  label[for="income"] {
+    border-radius: 30px 0 0 30px;
+    border: 0.1px solid var(--dark-grey-color);
+    background-color: white;
+    font-size: 14px;
+  }
+
+  label[for="expense"] {
+    border-radius: 0 30px 30px 0;
+    border: 0.1px solid var(--dark-grey-color);
+    background-color: white;
+    font-size: 14px;
+  }
+
+  input:checked + label {
+    background-color: var(--accent-color);
+    color: white;
+  }
+`;
+const StyledButton = styled.button`
+  border-radius: 24px;
+  background-color: white;
+  padding: 4px 16px;
+  border: 1px solid var(--dark-grey-color);
+  color: var(--text-color-dark);
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
+  margin-top: 12px;
```

Comment:
Is it necessary to use `margin` here? We were told to use as little `margin` as possible as it can cause problems.